### ISSUE TITLE
referencing equations inside `$$ \begin{align}` environment

### ISSUE THREAD

**Issue by HelenaLC** [quarto-dev/quarto-cli#2275]
### Bug description

I am struggling to label multiple equations placed within `$$ \begin{align} ... \end{align} $$`. 

This works flawlessly...
```
$$
\begin{align}
a &= 0+1 \\
b &= 2+3 \\
c &= 4+5
\end{align}
$$ {#eq-abc}

see @eq-abc.
```

...but neither of these work:

```
$$
\begin{align}
a &= 0+1 \label{eq:a} \\
b &= 2+3 \label{eq-c} \\
c &= 4+5 {#eq-d}
\end{align}
$$
```

I tried various other versions of Quarto and/or LaTeX syntax, but couldn't achieve the desired results: To have multiple numbered (and cross-referenceable) equations placed within an align-block. 

Any suggestions would be greatly appreciated! And, it would be nice to have such an example included in the Equations sections of the Technical Writing documentation [here](https://quarto.org/docs/visual-editor/technical.html).

**Software:**
- Quarto version 1.1.168
- macOS Monterey version 12.2
- RStudio IDE version "Prairie Trillium" Release  
(1db809b8, 2022-05-16) for macOS

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://yihui.org/issue/#please-format-your-issue-correctly) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by mattocci27** [comment]
Are there any workaround for MS word documents?

**Comment by abudden** [comment]
The workaround that uses `\label` and `\eqref` is probably better with `\ref`:

```
\begin{align}
a & = b + c \label{eq1} \\
c & = d + e \label{eq2}
\end{align}

Referring to equations \ref{eq1} and \ref{eq2}...
```

The difference is that `\eqref` renders the equation number as `(1)` where as `\ref` renders it as `1`.  Arguably the `\eqref` version is better, but it won't match the other equations in the document (ref #2439).

**Comment by eeholmes** [comment]
This
```
\begin{align}
a & = b + c \label{eq1} \\
c & = d + e \label{eq2}
\end{align}
```
will not render equation numbers in html. You could add `\tag{1}` but then you lose the autonumbering feature. The problem is that by default MathJax is set up to not show the numbers. Why the MathJax folks choose the opposite default as LaTeX makes no sense but that is what they did. So you have to turn on numbering. Sadly how you do this depends on the version of MathJax and the documentation is not 100% helpful. 🤯🤯

Here is how to turn on numbering with the MathJax js file that Quarto is using.

mathjax.html
```
<script>
  MathJax = {
    tex: {
      tags: 'ams'  // should be 'ams', 'none', or 'all'
    }
  };
</script>
```
Note the other common way is to set `MathJax.Hub.Config` so search for that if this doesn't work with the MathJax js file you are using.
:
_quarto.qmd yaml looks like this now
```
---
title: "test"
format:
  html:
    include-in-header: mathjax.html
  pdf: default
---
```

Now this shows your autonumbers in both html and pdf output.
```
\begin{align}
a & = b + c \label{eq1} \\
c & = d + e \label{eq2}
\end{align}
```

Something to keep in mind is that the Quarto equation numbering is incompatible with LaTeX equation environment numbering **in html** but works fine in PDF. Weird. Just don't use the (handy) @eq-eq1 because Pandoc doesn't figure that out for the equation environments. Look at this example that mixes `$$ $$` with equation environments (`\begin{align}`).
```
$$
S_n = \frac{X_1 + X_2 + \cdots + X_n}{n}
      = \frac{1}{n}\sum_{i}^{n} X_i
$$ {#eq-eq1}

\begin{align}
a & = b + c \label{eq2} \\
c & = d + e \label{eq3}
\end{align}

$$
S_n = \frac{X_1 + X_2 + \cdots + X_n}{n}
      = \frac{1}{n}\sum_{i}^{n} X_i
$$ {#eq-eq4}
```

Produces this in html:
<img width="898" alt="image" src="https://user-images.githubusercontent.com/2545978/213819430-91333c30-8db5-4c4b-a859-40c924a5f9b4.png">


Produces this in PDF
<img width="898" alt="image" src="https://user-images.githubusercontent.com/2545978/213819371-1d46146c-9428-48b8-afcc-5516e616cf9b.png">

**Comment by eeholmes** [comment]
Note I tried a mathjax.html file with both ways of turning on numbering, and it still worked. So this file may be a more robust way to turn on numbering since it will work with more versions of MathJax (I think).  

```
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: {autoNumber: "AMS"} },
  tex2jax: {inlineMath: [ ['$','$'], ["\\(","\\)"] ]}
});
</script>

<!-- This is what works with Quarto -->
<script>
  MathJax = {
    tex: {
      tags: 'ams'  // should be 'ams', 'none', or 'all'
    }
  };
</script>
```

But weirdly with the MathJax file that RStudio Rmd files use, the line up of the equations numbers changes. Here is the format spec
```
format:
  html:
    include-in-header: mathjax.html
    html-math-method:
      method: mathjax
      url: "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"
```

<img width="946" alt="image" src="https://user-images.githubusercontent.com/2545978/213821306-21a90dbe-74aa-4c06-8b78-5af158aacb44.png">

The MathJax js file that Quarto is using by default is:
```
"https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"
```

**Comment by pglpm** [comment]
@eeholmes Thank you *so much* for the mathjax config scripts and the detailed explanations. 

One question: the [Mathjax 3.2 doc pages](https://docs.mathjax.org/en/latest/input/tex/extensions.html) use `window.MathJax` at the beginning of configuration files. Is this equivalent to the `MathJax` at the beginning of the config scripts above? Or are these two completely different things? Cheers!

**Comment by ikosmidis** [comment]
Just landed on this thread, and here is a proposal using the `aligned` environment within $$ ... $$.

```
$$
\begin{aligned}
a &= 0+1 \\
b &= 2+3 \\
c &= 4+5
\end{aligned}
$$ {#eq-abc}

see @eq-abc, or ([-@eq-abc]).
```

This works with default settings with correct labels in both PDF and HTML ( and I guess word!) and does not require any MathJax hacking. 

Useful thing is that 
```
format:
  html:
    crossrefs-hover: true
```
will display referenced equations on hover in HTML (and PDF, depending on your PDF reader), which is handy with long technical content.

**Comment by asmaier** [comment]
It should be noted that using `aligned` instead of `align` won't work. This document `test.md`

```
---
title: "test"
format:
  html:
    include-in-header: mathjax.html
---

$$\begin{aligned}
a & = b + c \label{eq2} \\
c & = d + e \label{eq3}
\end{aligned}$$

Refer to `\eqref{eq2}`{=latex} and `\eqref{eq3}`{=latex}.
```
will render using `quarto preview test.md` like

![Bildschirmfoto 2024-04-19 um 23 53 26](https://github.com/quarto-dev/quarto-cli/assets/3169297/16923ecb-2b73-4bbf-9fae-09bb95d74a73)

Instead of two equations with two labels mathjax counts them as one. 

Why is this relevant? If one starts with the following latex file `test.tex` 

```
\documentclass[12pt,a4paper,twoside]{book}
\usepackage[intlimits]{amsmath}	

\begin{document}

\begin{align}
a & = b + c \label{eq2} \\
c & = d + e \label{eq3}
\end{align}

Refer to \eqref{eq2} and \eqref{eq3}.

\end{document}
```
and converts it to markdown via `quarto pandoc --wrap=preserve -t markdown -f latex+raw_tex  -o test.md test.tex` then pandoc will unfortunately convert the `align` environment into an `aligned` environment (because of this issue https://github.com/jgm/pandoc/issues/4104) and one ends up with a markdown file as shown above.

**Comment by ikosmidis** [comment]
My solution is correct for just a single label for all aligned equations in a single display. I tend not to need more than one label, mainly because I avoid more than three equations in a single display, and then I can always say the first / second / third / etc. equation/expression/ etc. in XYZ.

Labelling more than 1 equation in a single align display in the current quarto version seems impossible without the Mathjax hacks in previous responses. But with those, many nice goodies that quarto provides and guarantees on the outputs between formats may be lost.

**Comment by mtartari** [comment]
I tried to follow to the letter the instructions by @eeholmes but what I get in my HTML output is this:

![Screenshot 2024-07-25 at 12 53 39 AM](https://github.com/user-attachments/assets/8815aac4-9bb5-436f-8738-782b13a0c15a)

To be precise here is what I did:

1. saved a file named `mathjax.html` in my project's directory, it has the exact content copied+pasted from @eeholmes post, nothing more nothing less:

```
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: {autoNumber: "AMS"} },
  tex2jax: {inlineMath: [ ['$','$'], ["\\(","\\)"] ]}
});
</script>

<!-- This is what works with Quarto -->
<script>
  MathJax = {
    tex: {
      tags: 'ams'  // should be 'ams', 'none', or 'all'
    }
  };
</script>
```

2.  In my existing`_quarto.yml` file I have this (note: last 4 lines are copied+pasted from @eeholmes post):
```
format:
  html:
    theme: custom.scss
    css: style.css
    pandoc_args: ["--lua-filter=color-text.lua"]
    include-in-header: mathjax.html
    html-math-method:
      method: mathjax
      url: "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"
```

3. In `intro.qmd` I copied the LaTeX script:
```
\begin{align}
a & = b + c \label{eq1} \\
c & = d + e \label{eq2}
\end{align}
```

Perhaps I am misunderstanding @eeholmes instructions, I apologize in advance for my total ignorance (in particular regarding the discussion around the version of Mathjack). I appreciate the help because I teach and I need to label equations in a long derivation ALL the time and I just switched from Bookdown (where I could do that) to Quarto and bumped into this issue.

**Comment by eeholmes** [comment]
@mtartari Look at this repo https://github.com/nmfs-opensci/test-equations. You might be missing the `$$` fences.  I am using the latest version of Quarto 1.5.55 with RStudio.

<img width="855" alt="image" src="https://github.com/user-attachments/assets/4b42eef2-3ec9-476d-8dbf-edb4b9ced0c2">

**Comment by Scott-Akenhead** [comment]
Using \begin{align}  without $$ renders as HTML and as PDF. Here’s the test code and a screen-shot of PDF resulting.
For instance,
\begin{align}
y &= \mathcal{N} (\hat{y}, \  \sigma) \\
  &= \mathcal{N} (\alpha + X \beta + Z \gamma, \  \sigma) 
\end{align}  
Posit should, please, document this.

<img width="459" alt="image" src="https://github.com/user-attachments/assets/fe8a4929-e7af-4218-bd7c-2d42a7bb5506">

**Comment by eeholmes** [comment]
@Scott-Akenhead Yes it works if you don't have `\label{eq1}` (so that you can ref the equation lines) but does not work if you do have that.  If you do have that you'll see:

<img width="420" alt="image" src="https://github.com/user-attachments/assets/ea5fd483-2485-4d5f-a4f4-510316c8223f">

**Comment by Scott-Akenhead** [comment]
Sheesh. This is messy!  I tried your suggestion:

> \begin{equation*}
> \begin{aligned}
> y &= \mathcal{N} (\hat{y}, \  \sigma) \label{eqx}\\
>   &= \mathcal{N} (\alpha + X \beta + Z \gamma, \  \sigma) \label{eqy}
> \end{aligned}
> \end{equation*}  

but result was 

> ERROR: 
> compilation failed- error
> Package amsmath Error: Multiple \label's: label 'eqx' will be lost.                                                  
> l.154 ...a + X \beta + Z \gamma, \  \sigma) \label
>                                                   {eqy} 

Same result with the equations inside a tex chunk: 

> ```{=tex}
> \begin{equation*} \begin{aligned}
> y &= \mathcal{N} (\hat{y}, \  \sigma) \label{eq1} \\
> y &= \mathcal{N} (\alpha + X \beta + Z \gamma, \  \sigma) \label{eq2} 
> \end{aligned} \end{equation*}
> ```
> 

and even when equation number is automatic, without labels, there is only 1 equation number:

> ```{=tex}
> \begin{equation} \begin{aligned}
> y &= \mathcal{N} (\hat{y}, \  \sigma) \\
> y &= \mathcal{N} (\alpha + X \beta + Z \gamma, \  \sigma)  
> \end{aligned} \end{equation}
> ```
<img width="408" alt="image" src="https://github.com/user-attachments/assets/6871da19-3144-49cd-86b0-bfb4122c31d4">


**So hath the Latex Math God decreed and all men shall bow to his/her/its command.**

**Comment by eeholmes** [comment]
Try putting in $$ fence?

**Comment by eeholmes** [comment]
@Scott-Akenhead Something changed with Quarto since my Jan 20, 2023 answer. No to get `\begin{align}` to work I have to put it in a `$$` fence not `\begin{equation*}`. I realize that is improper LaTeX but that's what I had to do.

**Comment by cscheid** [comment]
I don't think we intentionally changed anything! If you folks can come up with a minimal document where the behavior changed, we will be able to more easily help.

**Comment by eeholmes** [comment]
@cscheid Sorry we didn't get to meet up at Posit conf! So close...

I am moving my various notes from this issue and another into a repo to chip away at the issues for LaTeX users. Also will provide me a place to send LaTeX users for examples. 

* https://github.com/nmfs-opensci/quarto-amsmath
* https://nmfs-opensci.github.io/quarto-amsmath/

I don't think it is any Quarto bug per se. My goal is to figure out what needs to be done to get proper amsmath LaTeX syntax working in Quarto books. https://github.com/nmfs-opensci/quarto-amsmath/issues/2.  I realize that Quarto syntax for equation numbering is working, but that is causing too many issues for LaTeX users who are trying to get their amsmath code working. So rather than try to create a new equation syntax, I'd like to figure how to get amsmath syntax working. It is pretty close. Most things work fine. It is just the section numbering that I don't know how to fix.

**Comment by Scott-Akenhead** [comment]
On behalf of all of us using Quarto to collaborate on science papers: 
Thanks! Appreciated!

An encompassing, over-the-horizon, goal might be: Submit a paper in Quarto _directly_ to a science journal.

**Comment by eeholmes** [comment]
@cscheid I added this comment to the issue I am working on. https://github.com/nmfs-opensci/quarto-amsmath/issues/2#issuecomment-2294989880 Maybe it will explain the problem better? Quarto is essentially circumventing the MathJax javascript libraries way of handling and figuring out equation numbering in our complicated equations. Recreating the MathJax library behavior would not be good. It's really complicated and took years of work to make complicated amsmath equations look right with numbers in the right places in html. 

The solution is not too hard. I have done it before. It uses the MathJax tagformat extension. Solution is [here](https://docs.mathjax.org/en/latest/input/tex/extensions/tagformat.html#tex-tagformat). But it involves adding `<span style="display: hidden">\(\nextSection\)</span>` to each chapter qmd. I want to figure out a way to do that automatically. A "mathjax lua filter" might work and feel natural to users (natural in the sense that using lua filters in quarto is natural).

**Comment by cscheid** [comment]
> A "mathjax lua filter" might work and feel natural to users (natural in the sense that using lua filters in quarto is natural)

This might be harder said than done. I have to spend some time investigating to say for sure, but I believe some amount of the processing for MathJax happens after Lua filters are done (some of it is done by Pandoc, and some of it is done by Quarto post-processors). At the point where the document is at a state that you want it to be, there might not be any Quarto extension entry points.

**Comment by eeholmes** [comment]
@cscheid Here is the repo showing where I am at. I am quite close to a 'proof of concept'. If I can just figure out how to id the `<header>` tags.

https://github.com/nmfs-opensci/chapnum-equations

**Comment by bjornasm** [comment]
Is it possible to read some proper documentation on how to handle referencing in equations? There are a lot of suggestions in this thread but I can't really understand which would be the right one.

**Comment by FelixBenning** [comment]
Is it realistic that quarto will eventually be able to make this work?
```tex
$$
\begin{aligned}
% | {#eq-a}
a &= 0+1  \\
% | {#eq-b} 
b &= 2+3 \\
% | {#eq-d}
c &= 4+5 
\end{aligned}
$$
```
This would be similar to how this comment then `| ` syntax is used in code snippets to send instructions to quarto. E.g.

````qmd
```{python}
# | echo: false
pass
```
```` 

Circumventing the native quarto referencing and relying on mathjax
- breaks on-hover previews for links in html
- breaks cross-page references that mathjax does not know, so you cannot reference an equation from a different chapter with the mathjax approach. Additionally you cannot get the equation tags to be formatted as `(chap.eq_num)`, e.g. `(3.2)` easily.

So I would really not recommend circumventing the quarto references. At the same time it would be nice to eventually be able to reference multi-line equations again.

**Comment by FelixBenning** [comment]
## Workaround: Case-by-case treatment

Typically, you do not need to reference individual lines in an equation.  The following workaround works well if you only need this in a handful of places and only reference these individual equations within the same chapter.
It works by duplicating the equation and providing a version for the `html` target separately for a version for the `pdf` (i.e. `LaTeX` target).

It looks like this in html

<img width="538" alt="The equation has one equation label (1.3) at the right margin and additionally the lable (a) for the first and the label (b) for the second line right after the content. Later on Equation 1.3 (a) is referenced" src="https://github.com/user-attachments/assets/3f84ac5a-fb8f-4db2-98fa-e17ad3a0a292" />

and like this in LaTeX

<img width="480" alt="The first line of the equation has label (1.3), the second line has label (1.4), later on Equation 1.3 is referenced" src="https://github.com/user-attachments/assets/e165d538-ac5a-4d8d-b1ac-f5245f1fea75" />

This is achieved using the following code that introduces a case-by-case treatment of the output

```qmd
::: {.content-visible when-format="html"}
$$
	\begin{align}
		a &= b \tag{a}\\
		&=c \tag{b}
	\end{align}
$$ {#eq-test}
:::
::: {.content-visible when-format="pdf"}
	\begin{align}
		a &= b \label{eq-a}\\
		&=c \label{eq-b}
	\end{align}
:::

`<` Text before you want to reference the equation. `>`
Equation reference:
[@eq-test (a)]{.content-visible when-format="html"}
[Equation \ref{eq-a}]{.content-visible when-format="pdf"}
```

Note that `\begin{align}` is invalid syntax inside the math environment for `LaTeX` but it is accepted by `mathjax`. This allows double tags for the `html` target. In `LaTeX` you cannot have these duplicate tags, but you can use the fact that Quarto converts equation references into the `LaTeX` reference system and `LaTeX` will then take all labels into consideration and the numbering simply works. Check both outputs!

**Comment by mcanouil** [comment]
@FelixBenning Accessibility: To improve accessibility, please add alternative text to your screenshots. This helps all users, including those using screen readers, to understand the context of the images. A brief description can make a big difference! See [Good Alt Text, Bad Alt Text — Making Your Content Perceivable](https://www.wcag.com/blog/good-alt-text-bad-alt-text-making-your-content-perceivable/).

**Comment by friendly** [comment]
I'm dealing with this very issue now in my book, where I have ~10-12 equations of the form

```
\begin{align*}
  ... equation stuff ...
\end{align*} {#eq-label}
```
so I want one eqn number for the set that I can references as `@eq-label` 

For a concrete example:

```
\begin{align*}
\mathbf{y} & = \beta_0 + \beta_1 \mathbf{x}_1 + \beta_2 \mathbf{x}_2 + \cdots + \beta_p \mathbf{x}_p + \boldsymbol{\epsilon} \\
           & = \left[ \mathbf{1},\; \mathbf{x}_1,\; \mathbf{x}_2,\; \dots ,\; \mathbf{x}_p \right] \; \boldsymbol{\beta} + \boldsymbol{\epsilon} \\
\end{align*} {#eq-glm}
```

In HTML, one looks like this:

<img width="841" height="169" alt="Image" src="https://github.com/user-attachments/assets/68d392c0-8afa-4d58-aad6-68dec213c62d" />

But in PDF I get no equation number and x-refs don't work:

<img width="990" height="265" alt="Image" src="https://github.com/user-attachments/assets/1b5876bc-1dbd-45d8-8b8d-05362f1d2aed" />


I've read through the discussion here, but not sure of a recommended solution. Do I have to resort to the **Case-by-case workaround** suggested by @FelixBenning ??

**Comment by FelixBenning** [comment]
If you never want to reference individual lines simply use the inbuilt `$$ <content>$$ {#eq-tag}` syntax

**Comment by friendly** [comment]
NO -- the problem is that (last I tested this), you cannot use `$$ \begin{align*} <equations> \end{align*} $$ {eq-tag}`, because the aligned environment already implies an equation environment -- that triggers an error.

**Comment by friendly** [comment]
Here's my test:

```
Try surrounding this in `$$`

$$
\begin{align*}
\mathbf{y} & = \beta_0 + \beta_1 \mathbf{x}_1 + \beta_2 \mathbf{x}_2 + \cdots + \beta_p \mathbf{x}_p + \boldsymbol{\epsilon} \\
           & = \left[ \mathbf{1},\; \mathbf{x}_1,\; \mathbf{x}_2,\; \dots ,\; \mathbf{x}_p \right] \; \boldsymbol{\beta} + \boldsymbol{\epsilon} \\
\end{align*} 
$${#eq-glm}
```

This works in HTML. But when I try to render in PDF, I get:

```
ERROR: 
compilation failed- error
Package amsmath Error: Erroneous nesting of equation structures;
(amsmath)                trying to recover with `aligned'.

See the amsmath package documentation for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.385 \end{align*}
```

**Comment by FelixBenning** [comment]
The `aligned' environment can be used in an equation environment. You can use that instead of align

**Comment by ikosmidis** [comment]
> Here's my test:
> 
> ```
> Try surrounding this in `$$`
> 
> $$
> \begin{align*}
> \mathbf{y} & = \beta_0 + \beta_1 \mathbf{x}_1 + \beta_2 \mathbf{x}_2 + \cdots + \beta_p \mathbf{x}_p + \boldsymbol{\epsilon} \\
>            & = \left[ \mathbf{1},\; \mathbf{x}_1,\; \mathbf{x}_2,\; \dots ,\; \mathbf{x}_p \right] \; \boldsymbol{\beta} + \boldsymbol{\epsilon} \\
> \end{align*} 
> $${#eq-glm}
> ```
> 
> This works in HTML. But when I try to render in PDF, I get:
> 
> ```
> ERROR: 
> compilation failed- error
> Package amsmath Error: Erroneous nesting of equation structures;
> (amsmath)                trying to recover with `aligned'.
> 
> See the amsmath package documentation for explanation.
> Type  H <return>  for immediate help.
>  ...                                              
>                                                   
> l.385 \end{align*}
> ```
> 

Hi @friendly: see https://github.com/quarto-dev/quarto-cli/issues/2275#issuecomment-1945766412 for how I got around it to get consistent referencing across output formats. The only caveat is that in this way you cannot cross-reference individual expressions, but I rarely need to in my research / lecture notes

**Comment by mcanouil** [comment]
I suggest reading the following resource which aggregates a lot about equations and Quarto:

- https://nmfs-opensci.github.io/quarto-amsmath/

**Comment by friendly** [comment]
@mcanouil that document on quarto-amsmath is no help, because it says you can't mix Quarto equation & amsmath syntax. Yet, most of my book uses Quarto syntax and it is only for equations where I need aligned equations that I'm forced to use amsmath. I view this as a quirky flaw in the general use of equations in Quarto/mathjax.

> tldr; For LaTeX users wanting complex math in both html and PDF: 1) Don’t mix Quarto equation and amsmath syntax. Stick with amsmath. 2) Add a mathjax.html file and add to your top yaml to turn-on equation numbering and only use LaTeX equation environments (not $$). 2) For referencing, only use \label{} and \eqref{} or \ref{}. Do not use (#eq-) and @eq- for cross-refs. Pandoc (and MathJax) knows LaTeX so all your fancy equations should work if you do 1 and 2.

@FelixBenning Gave me a **simple solution** for my use cases: use the amsmath `aligned` rather than `align`, because the former can be used inside `$$ ... $$`.  Here's my test case:

````
Or, try this with the `aligned` environment which can be used inside `$$ ... $$`

$$
\begin{aligned}
\mathbf{y} & = \beta_0 + \beta_1 \mathbf{x}_1 + \beta_2 \mathbf{x}_2 + \cdots + \beta_p \mathbf{x}_p + \boldsymbol{\epsilon} \\
           & = \left[ \mathbf{1},\; \mathbf{x}_1,\; \mathbf{x}_2,\; \dots ,\; \mathbf{x}_p \right] \; \boldsymbol{\beta} + \boldsymbol{\epsilon} \\
\end{aligned}
$${#eq-glm2}

As you see in @eq-glm2 ...
````

This gives me what I want:

<img width="730" height="151" alt="Image" src="https://github.com/user-attachments/assets/0d290178-1c11-41a7-a484-808998cddd21" />

**Comment by mcanouil** [comment]
I meant to look athttps://nmfs-opensci.github.io/quarto-amsmath/numbered_equations.html#basic-aligned although the code uses explicit `equation` environment, it can be replaced with `$$` exactly as suggested.

**Comment by friendly** [comment]
Hi @mcanouil 
That document uses straight LaTeX for equation labels via `\label{}` and then `\eqref{}` for references.  Maybe that also works for HTML, but my entire book uses Quarto refs `@eq-label`, I'm glad to have a solution that uses that. 

```
\begin{equation}  
\begin{aligned}
3(a-x) &= 3.5x + a - 1 \\
3a - 3x + 3a - 3x &= 3.5x + a - 1 
\end{aligned} \label{eq5}
\end{equation}

The cross-ref with \eqref{eq5} to get ...
```

**Comment by mcanouil** [comment]
> That document uses straight LaTeX 

That's a kind of a hidden Pandoc feature which works in Quarto, but that's not very robust as Quarto has its own cross-reference system.

Anyhow, unfortunately math environment is indeed a pain point which is a very hard one to fix thus the great number of workarounds.
